pkg/clusters/types/eks/aws-operations/eks.go

```diff
@@ -0,0 +1,479 @@
+	"github.com/samber/lo"
+	"github.com/weaveworks/eksctl/pkg/ami"
+	eksctlapi "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/authconfigmap"
+	eksiam "github.com/weaveworks/eksctl/pkg/iam"
+	"github.com/weaveworks/eksctl/pkg/nodebootstrap"
+	"k8s.io/client-go/kubernetes"
+	"os"
+	"time"
+)
+
+const (
+	DefaultNodeGroupName     = "default-node-group"
+	DefaultKubernetesSvcCIDR = "172.20.0.0/16"
+	kubernetesTagFormat      = "kubernetes.io/cluster/%s"
+	envKeyNodeSSHKeyName     = "EKS_NODE_SSH_KEY"
+
+	TagNameCreateBy = "ktf_created_by"
+)
+
+func CreateEKSClusterAll(ctx context.Context, cfg aws.Config, clusterName,
+	k8sMinorVersion, nodeMachineType string, tags map[string]string) error {
+
+	stsClient := sts.NewFromConfig(cfg)
+	ec2Client := ec2.NewFromConfig(cfg)
+	eksClient := eks.NewFromConfig(cfg)
+	iamClient := iam.NewFromConfig(cfg)
+
+	callIdentityOutput, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get caller identity")
+	}
+	createdByArn := aws.ToString(callIdentityOutput.Arn)
+
+	clusterRoleArn, nodeRoleArn, err := createRoles(ctx, iamClient, clusterName)
+	if err != nil {
+		return errors.Wrap(err, "failed to create IAM roles")
+	}
+	subnetAvZones, err := getAvailabilityZones(ctx, ec2Client, cfg.Region)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get availability zones in region %s", cfg.Region)
+	}
+
+	vpcId, subnetIDs, err := createVPC(ctx, ec2Client, subnetAvZones)
+	if err != nil {
+		return errors.Wrap(err, "failed to create VPC")
+	}
+
+	cpSgId, err := createControlPlaneSecurityGroup(ctx, ec2Client, vpcId, clusterName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create control plane security group in VPC %s", vpcId)
+	}
+
+	_, err = createCluster(ctx, eksClient, clusterName, clusterRoleArn, k8sMinorVersion, cpSgId, subnetIDs, createdByArn, tags)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create EKS cluster %s", clusterName)
+	}
+
+	activeCluster, err := waitForClusterActive(ctx, eksClient, clusterName)
+	if err != nil {
+		return errors.Wrapf(err, "failed while waiting for EKS cluster %s to become active", clusterName)
+	}
+
+	sgId, err := createNodeSecurityGroup(ctx, ec2Client, vpcId, clusterName, activeCluster.ResourcesVpcConfig.SecurityGroupIds)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create security groups")
+	}
+
+	_, kubeCfg, err := ClientForCluster(ctx, cfg, clusterName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get kube client for cluster %s", clusterName)
+	}
+
+	err = authorizeNodeGroup(kubeCfg, nodeRoleArn)
+	if err != nil {
+		return errors.Wrapf(err, "failed to authorize node group to access cluster %s", clusterName)
+	}
+
+	amiId, err := resolveAMI(ctx, ec2Client, cfg.Region, k8sMinorVersion, nodeMachineType, eksctlapi.DefaultNodeImageFamily)
+	if err != nil {
+		return errors.Wrap(err, "failed to resolve AMI")
```

Comment:
Can we print the Region or cluster name here?

Reply:
Cluster name is the environment/cluster-builder name, so it's known from outside.
Region is loaded from the environment variable as well and I'm open to print this. The concern is, this framework tends to not print anything during a cluster building. I'm not sure whether should I print or how should I log since it looks like we don't have a  logging mechanism provided in KTF.